pypesto/visualize/spline_approximation.py

```diff
@@ -218,6 +241,34 @@ def plot_splines_from_inner_result(
     return fig, axs
 
 
+def _calculate_optimal_regularization(
+    s: np.ndarray,
+    N: int,
+    c: np.ndarray,
+):
```

Comment:
docstring please

Reply:
with reference to your paper for the method, eventually